homeassistant/helpers/translation.py

```diff
@@ -32,24 +32,51 @@ def flatten(data: Dict) -> Dict[str, Any]:
 
 def component_translation_file(hass: HomeAssistantType, component: str,
                                language: str) -> str:
-    """Return the translation json file location for a component."""
-    if '.' in component:
-        name = component.split('.', 1)[1]
-    else:
-        name = component
+    """Return the translation json file location for a component.
+
+    For component one of:
+     - components/light/.translations/nl.json
+     - components/.translations/group.nl.json
+
+    For platform one of:
+     - components/light/.translations/hue.nl.json
+     - components/hue/.translations/light.nl.json
+    """
+    is_platform = '.' in component
+
+    if not is_platform:
+        module = get_component(hass, component)
+        assert module is not None
+
+        module_path = pathlib.Path(module.__file__)
 
-    module = get_component(hass, component)
+        if module.__name__ == module.__package__:
+            # light/__init__.py
+            filename = '{}.json'.format(language)
+        else:
+            # group.py
+            filename = '{}.{}.json'.format(component, language)
+
+        return module_path.parent / '.translations' / filename
+
+    # It's a platform
+    parts = component.split('.', 1)
```

Comment:
I would use:

```python
domain, platform = component.split('.', 1)
```

to avoid the use of parts[0] and parts[1]

Reply:
domain and platform can be the other way around now so that would be confusing 50% of the time